gammapy/datasets/flux_points.py

```diff
@@ -218,14 +238,21 @@ def read(cls, filename, name=None, format="gadf-sed"):
         )
 
     @classmethod
-    def from_dict(cls, data, **kwargs):
+    def from_dict(cls, data, format="gadf-sed", **kwargs):
```

Comment:
Maybe the possibility of setting it in the function call should be left open, in case the dict (for some reason) does not contain the format. I agree that the first choice should be read from dict.

It should be easy just adding a `if data.get("format")` statement at the start

Reply:
if the dict or yaml file does not provide the correct format and the user know about it then they should update the dict/ yaml file (rather than setting the format on read/from_dict)